types/tx_dynamic_fee.go

```diff
@@ -0,0 +1,93 @@
+package types
+
+import "math/big"
+
+// txDynamicFee implements Tx interface with the dynamic fee transaction logic.
+type txDynamicFee struct {
+	txBase
+}
+
+// SetNonce sets the given transaction nonce.
+func (t *txDynamicFee) SetNonce(nonce uint64) Tx {
+	t.nonce = nonce
+
+	return t
+}
+
+// SetGasPrice sets the given transaction gas price.
+func (t *txDynamicFee) SetGasPrice(gasPrice *big.Int) Tx {
+	t.gasPrice = new(big.Int).Set(gasPrice)
+
+	return t
+}
+
+// SetGas sets the given transaction gas.
+func (t *txDynamicFee) SetGas(gas uint64) Tx {
+	t.gas = gas
+
+	return t
+}
+
+// SetTo sets the given transaction receiver address.
+func (t *txDynamicFee) SetTo(addr Address) Tx {
```

Comment:
tx.Addr can be nil. It's using to create a contract. Using this signature we cant set To to nil.